lib/liquid.lua

```diff
@@ -2547,23 +2549,28 @@ end
 -------------------------------------------------------------ParserContext end ---------------------------------------------------------
 -------------------------------------------------------------FileSystem begin ---------------------------------------------------------
 -- local FileSystem = {}
-function FileSystem:new( location )
+function FileSystem:new( get )
     -- body
     local instance = {}
     setmetatable(instance, {__index = FileSystem})
-    instance.location = location
+    instance.get = get
     instance.text = nil
     return instance
 end
 --
-function FileSystem:genertic_get( ... )
+function FileSystem:genertic_get( location )
     -- body
     if self.get and type(self.get) == "function" then
-        local status, err = pcall(self.get, self.location)
-        if status then
-            return err
+        local ok, ret = pcall(self.get, location)
+
+	if not ret then
+            error("not found template " .. location)
```

Comment:
If the user's filesystem component get `nil` as its return value which will fail the Lexer component, it is totally reasonable for us to raise a error. Maybe a error msg like `error: template .. location:  return nil as content` will be better (we just want to point out  content `nil` is illegal, but not template is missing)